Use 'uv' as package manager

Bpolitycki opened the pull request:
- **feat: added orcid link**
- **feat: updated pyproject.toml**
- **feat: changed package-manager to uv**

# Pull request

## Proposed changes

<!-- A short description of the changes made in the PR. -->
This PR implements all changes needed to switch to `uv` instead of `poetry`. `uv` is the modern and much more standard alternative for managing python projects.
## Types of changes

<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] New feature (non-breaking change which adds functionality).
- [ ] Enhancement (non-breaking change which enhances functionality)
- [ ] Bug Fix (non-breaking change which fixes an issue).
- [x] Breaking change (fix or feature that would cause existing functionality to change).

## Checklist

<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->

- [x] I have read the **[README](./README.md)** document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

ChristianSonderUniSG committed:
feat: added orcid link

ChristianSonderUniSG committed:
feat: updated pyproject.toml

ChristianSonderUniSG committed:
feat: changed package-manager to uv

Bpolitycki committed:
fix: dependency versions

Bpolitycki committed:
ci: correct Taskfile-calls

ChristianSonderUniSG reviewed (changes requested):
Die Einbindung des SSRQ-CLI funktioniert immer noch nicht:

 ssrq-schema git:(uv) run install 
Installing dependencies...
Resolved 103 packages in 4ms
  × Failed to download and build `ssrq-cli @ git+ssh://git@github.com/SSRQ-SDS-FDS/ssrq-cli.git@26eaa1587633be822188aa245344a9eb845b4321`
  ├─▶ Git operation failed
  ╰─▶ process didn't exit successfully: `/opt/homebrew/bin/git reset --hard 26eaa1587633be822188aa245344a9eb845b4321` (exit status: 128)
      --- stderr
      Downloading tests/tex/examples/rdf-persons.xml (125 MB)
      Error downloading object: tests/tex/examples/rdf-persons.xml (62a30d2): Smudge error: Error downloading tests/tex/examples/rdf-persons.xml
      (62a30d235b9cb49cb37abbb2a1c8e3d55e5bb44b2551c08241229bcaff78efd4): error transferring "62a30d235b9cb49cb37abbb2a1c8e3d55e5bb44b2551c08241229bcaff78efd4":
      [0] remote missing object 62a30d235b9cb49cb37abbb2a1c8e3d55e5bb44b2551c08241229bcaff78efd4

      Errors logged to '/Users/christiansonder/.cache/uv/git-v0/checkouts/1c7e9b40de827d3e/26eaa15/.git/lfs/logs/20250908T105618.259038.log'.
      Use `git lfs logs last` to view the log.
      Fehler: externer Filter 'git-lfs filter-process' fehlgeschlagen
      Schwerwiegend: tests/tex/examples/rdf-persons.xml: smudge-Filter 'lfs' fehlgeschlagen

  help: `ssrq-cli` was included because `ssrq-schema` (v0.1.0) depends on `ssrq-cli`
Installing pre-commit hooks...
  × Failed to download and build `ssrq-cli @ git+ssh://git@github.com/SSRQ-SDS-FDS/ssrq-cli.git@26eaa1587633be822188aa245344a9eb845b4321`
  ├─▶ Git operation failed
  ╰─▶ process didn't exit successfully: `/opt/homebrew/bin/git reset --hard 26eaa1587633be822188aa245344a9eb845b4321` (exit status: 128)
      --- stderr
      Downloading tests/tex/examples/rdf-persons.xml (125 MB)
      Error downloading object: tests/tex/examples/rdf-persons.xml (62a30d2): Smudge error: Error downloading tests/tex/examples/rdf-persons.xml
      (62a30d235b9cb49cb37abbb2a1c8e3d55e5bb44b2551c08241229bcaff78efd4): error transferring "62a30d235b9cb49cb37abbb2a1c8e3d55e5bb44b2551c08241229bcaff78efd4":
      [0] remote missing object 62a30d235b9cb49cb37abbb2a1c8e3d55e5bb44b2551c08241229bcaff78efd4

      Errors logged to '/Users/christiansonder/.cache/uv/git-v0/checkouts/1c7e9b40de827d3e/26eaa15/.git/lfs/logs/20250908T105618.80737.log'.
      Use `git lfs logs last` to view the log.
      Fehler: externer Filter 'git-lfs filter-process' fehlgeschlagen
      Schwerwiegend: tests/tex/examples/rdf-persons.xml: smudge-Filter 'lfs' fehlgeschlagen

  help: `ssrq-cli` was included because `ssrq-schema` (v0.1.0) depends on `ssrq-cli`
Task completed in 0m1.147s

Bpolitycki committed:
chore: updated taskfile

The install cmd will now remove the `.venv`-dir at first and start a
fresh installation without using the cache.

ChristianSonderUniSG committed:
Update .pre-commit-config.yaml
